src/java_bytecode/java_object_factory.cpp

```diff
@@ -766,6 +766,13 @@ void java_object_factoryt::gen_nondet_pointer_init(
   // for java types, technical debt TG-2707
   if(!equal_java_types(replacement_pointer_type, pointer_type))
   {
+    // update generic_parameter_specialization_map if needed
```

Comment:
Consider elaborating this comment with an example. E.g.:
```
// if replacement_pointer_type is Implementation<T> implements I<T, Float>
// then this inserts pairs:
//   I::T -> Implementation::T, 
//   I::U -> java.lang.Float
```

Assuming my understanding is correct?

Reply:
Not quite. Here is an example of what happens here: 
1. Assume we see a pointer to `I<Integer>`.
2. `convert_pointer_type` finds a class `Impl<T,U> implements I<T>` and builds a replacement pointer `Impl<Integer, U>` (note that CBMC does not build these replacement pointers, only test generator).
3. Below we simply input the pairs for the replacement pointer to the map (nothing inheritance-specific): `Impl::T -> Integer` (the second parameter is unspecialized).